dhcp/dhcp.go

```diff
@@ -0,0 +1,202 @@
+package dhcp
```

Comment:
Hi Kjahed, just letting you know we are working on importing this code into Google internally, which has strict code quality standards. I'll be leaving comments based on what needs to be updated such that the code can pass our internal linters.

You may notice some of the already submitted code already violates the comments I leave, I'll be going through and gradually fixing everything up.

Reply:
Thanks @melzhan. I believe I addressed the ones you pointed out.

Reply:
Thanks @kjahed, code looks good to me now from a style perspective % typo. Leaving it to @gmacf to approve the logic.